magicblock-table-mania/src/manager.rs

```diff
@@ -0,0 +1,715 @@
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::{Duration, Instant},
+};
+
+use log::*;
+use magicblock_rpc_client::MagicblockRpcClient;
+use solana_pubkey::Pubkey;
+use solana_sdk::{
+    address_lookup_table::state::AddressLookupTable,
+    commitment_config::CommitmentConfig, message::AddressLookupTableAccount,
+    signature::Keypair, signer::Signer,
+};
+use tokio::{
+    sync::{Mutex, RwLock},
+    time::sleep,
+};
+
+use crate::{
+    error::{TableManiaError, TableManiaResult},
+    lookup_table_rc::{LookupTableRc, MAX_ENTRIES_AS_PART_OF_EXTEND},
+};
+
+// -----------------
+// GarbageCollectorConfig
+// -----------------
+
+/// Configures the Garbage Collector which deactivates and then closes
+/// lookup tables whose pubkeys have been released.
+#[derive(Debug, Clone)]
+pub struct GarbageCollectorConfig {
+    /// The interval at which to check for tables to deactivate.
+    pub deactivate_interval_ms: u64,
+    /// The interval at which to check for deactivated tables to close.
+    pub close_interval_ms: u64,
+}
+
+impl Default for GarbageCollectorConfig {
+    fn default() -> Self {
+        Self {
+            deactivate_interval_ms: 1_000,
+            close_interval_ms: 5_000,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct TableMania {
+    pub active_tables: Arc<RwLock<Vec<LookupTableRc>>>,
```

Comment:
I guess some research was made to justify the usage of async synchronization primitives, i.e. they will be used in highly concurrent environments and with lock guards being held across await points, if not, then their use is unjustified.